### PR TITLE
WiP: Create faq.md

### PR DESCRIPTION
**Pull request by samgnaniah**
## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: website/two-column-pages/docs/learn/faq.md [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+The memory footprint depends on the use case and the load. A simple Ballerina service which simultaneously serves 500 concurrent users will have a low memory footprint of  50 MB. A ballerina main() program with simple logic (such as a loop counter) consumes 15  MB.
+
+### What is the update process for Ballerina? What is the update frequency?
+
+We will update Ballerina once / month with a community release. These releases will require a roll-forward support requirement. Ballerina will have levels of language backwards compatibility commitments so that adopters can depend upon stability. When WSO2 releases commercial support we will have a binary patch model that works against older versions of Ballerina.
+
+### How do I report security vulnerabilities? 
+
+See http://ballerina.io/security.
```
Comment (samisawso2) [inline website/two-column-pages/docs/learn/faq.md]: The pointed page says:
Vulnerable WSO2 products(s) and version(s)
Should be 
Vulnerable Ballerina artifacts(s) and version(s)

File: website/two-column-pages/docs/learn/faq.md [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+
+Ballerina does not currently have an MVC framework. We’d love for the community to add one and publish on Ballerina Central. Ballerina’s object and struct model makes it easy to synchronize data and models. 
+
+### Is this more functional than object oriented?
+
+Ballerina is a modern language that attempts to bring together productive features of functional, imperative, and object-oriented programming.
+ 
+### Is Ballerina an object-oriented language?
+
+Both objects (types) and functions are first class constructs in Ballerina. Developers have the flexibility to choose the most appropriate option depending on the requirement.
```
Comment (samisawso2) [inline website/two-column-pages/docs/learn/faq.md]: Should we make objects and functions words appear as keywords in markup?

File: website/two-column-pages/docs/learn/faq.md [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+
+Ballerina is based on type equivalence, rather than type inheritance. The type system in ballerina is based on set theory and, therefore, type equivalence has more meaning for this domain than type inheritance.
+
+### Why does Ballerina not support overloading of methods and operators?
+
+This is for simplicity purposes.
+
+In Ballerina, functions can be defined with required parameters, defaultable parameters, and optional rest parameters. Ballerina supports calling defaultable parameters in any order by passing explicit `name=value` on invocation, and this approach cannot be combined with parameter overloading. We felt that this model offered package designers and developers more flexibility than overloading. 
+
+Ballerina also supports the use of function name as a function pointer. This makes it easy to understand code because there is a single function corresponding to a given function name and is more powerful than the use of function overloading.
```
Comment (samisawso2) [inline website/two-column-pages/docs/learn/faq.md]: Should we point to or provide here itself the syntax for the use of function pointers?

File: website/two-column-pages/docs/learn/faq.md [hunk 4]
```diff
@@ -0,0 +1,252 @@
...
+
+The linker in the Ballerina toolchain creates statically-linked binaries by default. All Ballerina binaries therefore include the Ballerina runtime, along with the runtime type information to support type checks, panic-time stack traces, observability metrics.
+
+A simple Ballerina “hello, world” program compiled and linked statically is around 600 kB.
+
+### What is the concurrency/threading model?
+
+Ballerina's runtime implements a virtual machine that executes the Ballerina bytecode, which is called the BVM. 
+
+The BVM's execution model is made up of lightweight parallel execution units known as "workers". A worker is a lightweight parallel execution unit. Every function or network action defines one or more workers to execute its logic, and each has its own lifecycle consisting of worker states such as READY, RUNNING, and WAITING_FOR_RESPONSE.
```
Comment (samisawso2) [inline website/two-column-pages/docs/learn/faq.md]: Do we have an LC diagram that we can add here?

File: website/two-column-pages/docs/learn/faq.md [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+
+This behavior gives a more natural programming environment for the developer, so she does not have to explicitly consider non-blocking I/O handling semantics. Also, this style of physical thread allocation is efficient and lowers the number of context switches optimizing CPU allocations.
+
+### Would you provide NLP, ML toolkits around Ballerina?
+
+We do not have any plans. The community can add their own packages through Ballerina Central, and we will work hard to delegate frameworks and projects to the community.
+
+### What enterprise patterns, microservices patterns does Ballerina support? 
+
+See the [Ballerina By Guide](/learn/guides/) to find out how to use Ballerina to do full lifecycle integration development. Many enterprise patterns are demonstrated with Ballerina as an example.
```
Comment (samisawso2) [inline website/two-column-pages/docs/learn/faq.md]: Did we mean to say BBE in the second sentence?

File: website/two-column-pages/docs/learn/faq.md [hunk 6]
```diff
@@ -0,0 +1,252 @@
...
+
+Ballerina’s concurrency model is built on the sequence diagram metaphor and offers simple constructs for writing concurrent programs. Its type system is a modern type system designed with sufficient power to describe data that occurs in distributed applications. It also includes a distributed security architecture to make it easier to write applications that are secure by design.
+
+Ballerina is designed for modern development practices with a modularity architecture based on packages that are easily shared widely. Version management, dependency management, testing, documentation, building and sharing are part of the language design architecture and not left for later add-on tools.
+
+The Ballerina standard library is in two parts: the usual standard library level functionality (akin to libc) and a standard library of network protocols, interface standards, data formats, authentication/authorization standards that make writing secure, resilient distributed applications significantly easier than with other languages.
+
+Ballerina has been inspired by Java, Go, C, C++, Rust, Haskell, Kotlin, Dart, Typescript, Javascript, Swift and other languages.
```
Comment (samisawso2) [inline website/two-column-pages/docs/learn/faq.md]: Should we copy the latest list from footer here?

Comment (samgnaniah) [reply website/two-column-pages/docs/learn/faq.md]: Done.

File: website/two-column-pages/docs/learn/faq.md (hunk 1 above)
Comment (samgnaniah) [reply website/two-column-pages/docs/learn/faq.md]: Addressed with this PR: https://github.com/ballerina-platform/ballerina-www/pull/585

File: website/two-column-pages/docs/learn/faq.md (hunk 4 above)
Comment (samgnaniah) [reply website/two-column-pages/docs/learn/faq.md]: Not that I know of. Will have to check and see or draw one.

File: website/two-column-pages/docs/learn/faq.md (hunk 5 above)
Comment (samgnaniah) [reply website/two-column-pages/docs/learn/faq.md]: Seem that way. Updated.

File: website/two-column-pages/docs/learn/faq.md (hunk 3 above)
Comment (samgnaniah) [reply website/two-column-pages/docs/learn/faq.md]: We could. What is the syntax used?

Comment (samgnaniah) [reply website/two-column-pages/docs/learn/faq.md]: Will point to this: https://stage.ballerina.io/learn/by-example/function-pointers.html

**Pull request merged**